pkg/osvscanner/vulnerability_result.go

```diff
@@ -109,12 +98,32 @@ func buildVulnerabilityResults(
 					includePackage = true
 				}
 			}
+
 			if actions.ScanLicensesSummary {
-				pkg.Licenses = licensesResp[i]
+				pkg.Licenses = psr.Licenses
 			}
 		}
 		if includePackage {
-			groupedBySource[rawPkg.Source] = append(groupedBySource[rawPkg.Source], pkg)
+			var sourceType string
+			switch p.SourceType {
+			case imodels.SourceTypeOSPackage:
+				sourceType = "os"
+			case imodels.SourceTypeProjectPackage:
+				sourceType = "lockfile"
+			case imodels.SourceTypeSBOM:
+				sourceType = "sbom"
+			case imodels.SourceTypeGit:
+				sourceType = "git"
+			case imodels.SourceTypeUnknown:
+				sourceType = "unknown"
+			default:
+				sourceType = "unknown"
```

Comment:
does it make sense to just have a single `default` that also covers the `SourceTypeUnknown` case?

Reply:
It does, but it fails the linter since we no longer cover all the enum cases. Maybe it should panic at default since source type should always be valid here, but having it be unknown doesn't hurt.